packages/backend-common/src/util/KubernetesContainerRunner.test.ts

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CoreV1Api,
+  KubeConfig,
+  RbacAuthorizationV1Api,
+} from '@kubernetes/client-node';
+import {
+  KubernetesContainerRunner,
+  KubernetesContainerRunnerOptions,
+} from './KubernetesContainerRunner';
+import { RunContainerOptions } from './ContainerRunner';
+import { PassThrough } from 'stream';
+
+const describeIfKubernetes = Boolean(process.env.KUBERNETES)
```

Comment:
Think it would be good to have comment here explaining the setup, how to run it, etc.

On a separate note, perhaps something like `KUBERNETES_TESTS`? just to convey a bit more intent with the naming